src/borg/remote.py

```diff
@@ -1220,8 +1209,7 @@ def unpack(data):
             csize, checksum = cache_struct.unpack(data[:cache_struct.size])
             compressed = data[cache_struct.size:]
             if checksum != xxh64(compressed):
-                logger.warning('Repository metadata cache: detected corrupted data in cache!')
-                raise RepositoryCache.InvalidateCacheEntry
+                raise IntegrityError('detected corrupted data in metadata cache')
```

Comment:
The same problem affects the encrypted cache. Bit flip in cache -> IntegrityError. So a chunk killer that uses a caching wrapper would always need to check whether the IE was caused by the cache.